src/main/java/org/prebid/server/bidder/pulsepoint/PulsepointBidder.java

```diff
@@ -1,89 +1,178 @@
 import org.prebid.server.proto.openrtb.ext.request.pulsepoint.ExtImpPulsepoint;
 import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class PulsepointBidder extends OpenrtbBidder<ExtImpPulsepoint> {
+public class PulsepointBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpPulsepoint>> PULSEPOINT_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
 
     public PulsepointBidder(String endpointUrl, JacksonMapper mapper) {
-        super(endpointUrl, RequestCreationStrategy.SINGLE_REQUEST, ExtImpPulsepoint.class, mapper);
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
     }
 
     @Override
-    protected Imp modifyImp(Imp imp, ExtImpPulsepoint extImpPulsepoint) throws PreBidException {
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> modifiedImps = new ArrayList<>();
+        String publisherId = null;
+
+        for (Imp imp : bidRequest.getImp()) {
+            final ExtImpPulsepoint extImpPulsepoint;
+            try {
+                extImpPulsepoint = parseImpExt(imp);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+                continue;
+            }
+
+            final Imp modifiedImp = modifyImp(imp, extImpPulsepoint);
+            modifiedImps.add(modifiedImp);
+
+            final Integer extPublisherId = extImpPulsepoint.getPublisherId();
+            if (publisherId == null && extPublisherId != null && extPublisherId > 0) {
+                publisherId = extPublisherId.toString();
+            }
+        }
+
+        if (modifiedImps.isEmpty()) {
+            return Result.withErrors(errors);
+        }
+        publisherId = Objects.requireNonNullElse(publisherId, StringUtils.EMPTY);
+
+        final BidRequest modifiedRequest = modifyRequest(bidRequest, publisherId, modifiedImps);
+        return Result.of(Collections.singletonList(createHttpRequest(modifiedRequest)), errors);
+    }
+
+    private ExtImpPulsepoint parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), PULSEPOINT_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private static Imp modifyImp(Imp imp, ExtImpPulsepoint extImpPulsepoint) {
         return imp.toBuilder().tagid(Objects.toString(extImpPulsepoint.getTagId())).build();
```

Comment:
`Objects.toString(extImpPulsepoint.getTagId())` can be simplified to `extImpPulsepoint.getTagId().toString()` as tagId cant be null, according to json schema.